api/envoy/api/v2/core/base.proto

```diff
@@ -76,6 +76,35 @@ message Locality {
   string sub_zone = 3;
 }
 
+// Envoy uses SemVer (https://semver.org/). Major/minor versions indicate
+// expected behaviors and APIs, the patch version field is used only
+// for security fixes and can be generally ignored.
+message SemanticVersion {
+  uint32 major = 1;
+  uint32 minor = 2;
+  uint32 patch = 3;
+}
+
+// Version and identification for an Envoy extension.
+message Extension {
+  // This is the name of the Envoy filter as specified in the Envoy
+  // configuration, e.g. envoy.router, com.acme.widget.
+  string name = 1;
+
+  // Category of the extension
+  string category = 2;
```

Comment:
Yes, I suspect that we will want gRPC clients to declare extensions in the future.

I assume that this field is intended to indicate whether the extension is (e.g.) an HTTP filter or an LB policy.  Are there any examples of categories that do not correspond with extension points in the xDS APIs?  If not, then an enum is probably fine, because the enum is more about the API than it is about Envoy specifically, and it should work fine for gRPC as well.

Reply:
Extension categories are conjoured by the REGISTER_FACTORY macro. I'm not aware of any centralized management of this, so I don't think it could be represented by an enum.

Reply:
I don't see why it couldn't be an enum.  It's true that we'd need to add a new value to the enum whenever we add a new type of extension, but presumably that doesn't happen very often.  And we could even have some sort of presubmit check that ensures that we don't forget to update the enum.

But this only makes sense if the categories of extensions correspond to extension points in the xDS APIs, which is why I asked about that above.  Is that the case?

Reply:
AFAIK there is no formal definition of an extension point. A 3rd party module can register arbitrary categories with arbitrary names.

For example, the "Dubbo" extension registers these categories: "dubbo_proxy.filters", "dubbo_proxy.protocols", "dubbo_proxy.route_matchers", "dubbo_proxy.serializers".

Reply:
I see -- so extensions themselves can create new categories, meaning that the set of possible values here is definitely not known a priori.

In that case, I do think that a reverse-DNS-style notation would be a good idea, so that we don't have different extensions stepping on each other.

Reply:
Yes, good point @jpeach, let's go with reverse DNS as @markdroth suggests. I think we can allow some distinguished for client builtins e.g. `envoy.<foo>`, `grpc.<foo>`. This matches the reverse DNS style elsewhere in xDS today.

Reply:
@yanavlasov can you document this convention here? See other example of reverse DNS explanation in the API.

Reply:
done

Reply:
The comments here don't describe a reverse-DNS syntax, they describe the current categories used in the Envoy implementation.  I think this needs to be changed, since the Envoy implementation won't work for xDS clients other than Envoy.